pkg/webui/oauth/views/update-password/index.js

```diff
@@ -0,0 +1,209 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { Container, Col, Row } from 'react-grid-system'
+import bind from 'autobind-decorator'
+import { defineMessages } from 'react-intl'
+import { push, replace } from 'connected-react-router'
+import { connect } from 'react-redux'
+import * as Yup from 'yup'
+import queryString from 'query-string'
+
+import api from '../../api'
+import sharedMessages from '../../../lib/shared-messages'
+import { digit, uppercase, special } from '../../lib/regexp'
+
+import Button from '../../../components/button'
+import Form from '../../../components/form'
+import Input from '../../../components/input'
+import SubmitButton from '../../../components/submit-button'
+import IntlHelmet from '../../../lib/components/intl-helmet'
+import Message from '../../../lib/components/message'
+import style from '../create-account/create-account.styl'
+import Checkbox from '../../../components/checkbox'
+import Spinner from '../../../components/spinner';
+
+const m = defineMessages({
+  newPassword: 'New Password',
+  oldPassword: 'Temporary or Old Password',
+  passwordChanged: 'Password changed successfully',
+  revokeAllAccess: 'Log out from all devices',
+  revokeWarning: 'This will revoke access from all logged in devices',
+})
+
+const validationSchema = Yup.object().shape({
+  password: Yup.string()
+    .min(8)
+    .matches(digit, sharedMessages.validatePasswordDigit)
+    .matches(uppercase, sharedMessages.validatePasswordUppercase)
+    .matches(special, sharedMessages.validatePasswordSpecial)
+    .required(sharedMessages.validateRequired),
+  confirm: Yup.string()
+    .oneOf([ Yup.ref('password'), null ], sharedMessages.validatePasswordMatch)
+    .min(8)
+    .required(sharedMessages.validateRequired),
+})
+
+const initialValues = {
+  password: '',
+  confirm: '',
+  password_changed: false,
+  revoke_all_access: false,
+}
+
+@connect(state => ({
+  fetching: state.user.fetching,
+  user: state.user.user,
+  }), {
+    push,
+    replace,
+})
+@bind
+export default class UpdatePassword extends React.PureComponent {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      error: '',
+      info: '',
+      revoke_all_access: false,
+    }
+  }
+
+  handleRevokeAllAccess (evt) {
+    this.setState({ revoke_all_access: evt.target.checked })
+  }
+
+  async handleSubmit (values, { setSubmitting }) {
+    const { user } = this.props
+    const userParams = queryString.parse(this.props.location.search)
+    const old_or_temp = values.old_password ? values.old_password : userParams.current
```

Comment:
please use snake and camel case consistently.
we usually keep using snake case when destructing models returned by the backend. the rest is camel case